src/sickle.ts

```diff
@@ -159,32 +223,83 @@ class Annotator {
   }
 
   private emitFunctionType(fnDecl: ts.FunctionLikeDeclaration) {
-    // The first \n makes the output sometimes uglier than necessary,
-    // but it's needed to work around
-    // https://github.com/Microsoft/TypeScript/issues/6982
-    this.emit('\n/**\n');
-    let existingAnnotation = this.existingClosureAnnotation(fnDecl);
-    if (existingAnnotation) {
-      this.emit(' * ' + existingAnnotation + '\n');
+    // Construct the JSDoc comment by reading the existing JSDoc, if
+    // any, and merging it with the known types of the function
+    // parameters and return type.
+    let jsDoc = this.getJSDoc(fnDecl);
+    let newDoc: JSDocComment = {tags: []};
```

Comment:
why the distinction of `newDoc` and `jsDoc`? You could just augment `jsDoc`, right?